artemis-commons/src/main/java/org/apache/activemq/artemis/logs/AuditLogger.java

```diff
@@ -2833,4 +2833,30 @@ static void isAutoCreated(Object source) {
    @LogMessage(level = Logger.Level.INFO)
    @Message(id = 601747, value = "User {0} is getting auto-created property on target resource: {1} {2}", format = Message.Format.MESSAGE_FORMAT)
    void isAutoCreated(String user, Object source, Object... args);
+
+   static void purge(Object source) {
+      LOGGER.purge(getCaller(), source);
+   }
+
+   @LogMessage(level = Logger.Level.INFO)
+   @Message(id = 601748, value = "User {0} is purging target resource: {1} {2}", format = Message.Format.MESSAGE_FORMAT)
+   void purge(String user, Object source, Object... args);
+
+
+   static void purgeAddressSuccess(String queueName) {
+      RESOURCE_LOGGER.purgeAddressSuccess(getCaller(), queueName);
+   }
+
+   @LogMessage(level = Logger.Level.INFO)
+   @Message(id = 601749, value = "User {0} has purged address {1}", format = Message.Format.MESSAGE_FORMAT)
+   void purgeAddressSuccess(String user, String queueName);
+
+
+   static void purgeAddressFailure(String queueName) {
+      RESOURCE_LOGGER.purgeAddressFailure(getCaller(), queueName);
+   }
+
+   @LogMessage(level = Logger.Level.INFO)
```

Comment:
log the purgeAddressFailure with a a warning with the exception details could help investigate on failures because retrieving the exception details from console isn't immediate.

Reply:
I'm not sure that's appropriate for the audit log. There are *lots* of places where failures are logged in the audit log and there are no debugging details. This is just following that same pattern (e.g. from `pause`, `resume`, etc.).